### DISCUSSION TITLE
button release after pulsetime expired

### DISCUSSION THREAD

**Discussion by Sundancescore** [arendst/Tasmota#15653] [Q&A] [answered]
Hello! First of all thank you for this wonderful, complex software package!

I use Tasmota on a Sonoff Mini and configure it with the followings:

SwitchMode 13
PulseTime 10

Everything works fine from web. But when I use the Button wired to the "manual comand" something unwanted is happening.
1. Everything is OK if I push the button and release it before pulsetime expires (that is: push the button and release it in less than a second). The relay is switched on, kept on for 1s and than switched off.
2. But if I push the button and keep it pushed more than 1s (after pulsetime expires) the relay is switched on, kept on for 1s until pulsetime expires, than I release the button and the relay is once again switched on. This is not what I want or expect.

I use the relay for a garage door to "emulate" the remote control. I use web commands from my mobile and also the manual switch (physical button hidden in a hard to find point). So, if I press the button more than a second the door will start to open (relay on / off according to pulsetime) and than when I release the button once again the relay is On and the door stops.

Is there a way to disable the relay to go on if I release the button after pulsetime expired?

**Comment by sfromis** [comment, marked as answer]
Why do you use SwitchMode 13 for that? In a simple test, SwitchMode 3 seems to handle the button being kept pushed.

**Comment by Sundancescore** [comment]
I use SwitchMode 13 because I have to use PulseTime. The garage door opens to a "pulse" (like a short press/release of a button on a Remote control). I don't need the relay to stay on, but just to give a pulse to the motherboard that handles the garage door.

**Reply by sfromis** [reply]
That's no answer to why you selected to use SwitchMode 13 instead of what appears to work.

**Comment by Sundancescore** [comment]
I also use a HTML page also to control the garage door as mentioned ("I use web commands from my mobile and also the manual switch"). From the web page I send the On command (Power%20On) that triggers the Relay On and than PulseTime setting switches the relay to Off automatically. This emulates the press/release like a remote button for the web application. 

The only mode that works with PulseTime is SwitchMode 13 as far as I know.

**Reply by sfromis** [reply]
As I wrote, I made a small test with SwitchMode 3, which appeared to handle the situation of keeping the button pressed. Why does this not work for you? PulseTime is certainly not only for SwitchMode 13.

Or you could also config it as a Button, which may be simpler when it is not a toggle switch.

**Reply by Sundancescore** [reply]
That you for insisting with SwitchMode 3! :) This is really working as expected.

I was under the impression that PulseTime works only with SwitchMode 13 because in the docs this is the only mode where PulseTime is mentioned. In fact it works like a charm with Switchmode 3.

**Comment by barbudor** [comment]
I'm not at my desk and can't test for now but As I see it, Pulsetime react to an Power ON event, not directly to a button or switch event.

With switchmode 3, you should get a TOGGLE event when pressed followed by a OFF event when release.
I haven't tested it right now but I would assume that with a long enough Pulsetime, the press would power on and start the pulse timer, but then the release would trigger a Off, breaking the pulse timer.

With switchmode 13, as no event is generated on release that makes the switch most suitable for a pulse generator

**Reply by Sundancescore** [reply]
I did the tests with SwitchMode 3 and releasing the button after PulseTime expires does nothing. So everything works as expected, thank you for your pointing out this possible issue.

**Reply by barbudor** [reply]
I was pointing at release before the pulsetime expired

**Reply by Sundancescore** [reply]
I tried that also. Works fine. Nothing happens at release (off) before PulseTime expires. The relay goes off as expected when PulseTime expires.

Thank you for your time! :)
